### ISSUE TITLE
Early Stopping in DIGITS

### ISSUE THREAD

**Issue by JakaCikac** [NVIDIA/DIGITS#1729]
Hello there,

is there a way to use **early stopping** in DIGITS to prevent overfitting or stop training automatically when there is an increase in the validation loss to not waste time with the model? 

If there is no such a thing what would be a good way of implementing it and where would be a good place (in the code) to plug it in? 

Thank you and kind regards,
Jaka

**Comment by ethantang95** [comment]
As far as I know, there is no way to do that. But the best way to have a behavior to something like that is to use the DIGITS API and set up an application to receive the output and then send back a request based on it. Hope this helps. Thank you very much

**Comment by JakaCikac** [comment]
Hello!

Sounds like a good idea. I will look into it.
It would be nice if DIGITS would support it in the GUI itself thought! Also, I was thinking of checking the validation loss where the data is gathered to create the chart during training and then send an abort to the job when overfitting is "detected". 

Thanks for your answer!

**Comment by mrgloom** [comment]
Any update on this?

Also in Caffe itself seems you need to recompile a source to add early stopping functionality.
https://stackoverflow.com/questions/40912242/early-stopping-in-caffe
